lib/ui/painting.dart

```diff
@@ -1418,6 +1418,44 @@ class Canvas extends NativeFieldWrapperClass2 {
   /// entire group can be made transparent using the [saveLayer]'s paint.
   ///
   /// Call [restore] to pop the save stack and apply the paint to the group.
+  ///
+  /// ## Using saveLayer with clips
+  ///
+  /// When a rectanglular clip operation (from [clipRect]) is not axis-aligned
+  /// with the display, or when the clip operation is not rectalinear (e.g.
+  /// because it is a rounded rectangle clip created by [clipRRect] or an
+  /// arbitrarily complicated path clip created by [clipPath]), the edge of the
+  /// clip needs to be anti-aliased.
+  ///
+  /// If two draw calls overlap at the edge of such a clipped region, without
+  /// using [saveLayer], the first drawing will be anti-aliased with the
+  /// background first, and then the second will be anti-aliased with the result
+  /// of blending the first drawing and the background. On the other hand, if
+  /// [saveLayer] is used immediately after establishing the clip, the second
```

Comment:
I think the mental model is that that the clip is applied when you call `restore()`.  Specifically, it's part of the pipeline of transformations (e.g., filters, clips, blends, etc) that is applied to the temporary buffer as it is copied into the parent buffer.  Conceptually, it would probably make more sense to combine the call with the saveLayer call, as the filters and blends are, but that's not how the Skia API is set up.